src/v/cloud_storage/topic_mount_manifest.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#pragma once
+
+#include "cloud_storage/base_manifest.h"
+#include "cloud_storage/fwd.h"
+#include "cloud_storage/types.h"
+#include "model/fundamental.h"
+#include "model/metadata.h"
+
+#include <seastar/core/future.hh>
+
+#include <optional>
+
+namespace cloud_storage {
+
+class topic_mount_manifest final : public base_manifest {
+public:
+    topic_mount_manifest(
+      const model::cluster_uuid& uuid,
+      const model::topic_namespace& tp_ns,
+      const std::optional<ss::sstring>& recovery_prefix = std::nullopt);
+
+    ss::future<> update(ss::input_stream<char> is) override;
+
+    ss::future<serialized_data_stream> serialize() const override;
+
+    ss::sstring get_recovery_prefix() const;
+
+    const model::cluster_uuid& get_uuid() const { return _uuid; }
+
+    const model::topic_namespace& get_tp_ns() const { return _tp_ns; }
+
+    /// Manifest object name in S3
+    remote_manifest_path
+    get_manifest_path(const remote_path_provider& path_provider) const;
+
+    manifest_type get_manifest_type() const override;
+
+private:
+    model::cluster_uuid _uuid;
```

Comment:
I'd suggest making this a `remote_label` instead of a plain uuid. My intent with remote label is that if we ever want to further disambiguate topics with something in addition to just cluster UUID, it'd affect all paths.

Reply:
Also it may be obvious, but I think it'd be helpful to explicitly describe this as the uuid belonging to the source topic. Especially if/when we get around to implementing migrations where we have a target destination UUID, in which case maybe we would include the target destination in the body of the manifest

Reply:
Changed `_uuid` in `topic_mount_manifest` to a `cloud_storage::remote_label _source_label;`, and added a descriptive comment.

```cpp
// The label for the cluster that originally created the topic to be
// mounted/unmounted.
remote_label _source_label;
```